pca_cca.go

```diff
@@ -0,0 +1,252 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import (
+	"errors"
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix"
+	"github.com/gonum/matrix/mat64"
+)
+
+// PrincipalComponents returns the principal component direction vectors and
```

Comment:
How about replacing this with
```
// PrincipalComponents returns the principal component vectors and the variances
// of the principal component scores of the input data which is represented as
// an n×d matrix a where each row is an observation and each column is a variable.
//
// The slice weights is used to weight the observations. If weights is nil, each
// weight is considered to have a value of one, otherwise the length of weights
// must match the number of observations or PrincipalComponents will panic.
//
// On successful completion, the principal component vectors are returned in
// the columns of vecs as a d×min(n,d) matrix, and the column variances of
// the principal component scores, vecs * a^T, are returned in vars as
// a min(n,d)-long slice in descending sort order.
// 
// On failure, vecs and vars are returned nil and ok is returned false.
```
I omitted the reference to SVD and the remark about centering and not scaling. Feel free to put it back. 

I also fixed the missing transpose in `vecs * a^T` (please check that I'm not wrong).